apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/compute/functions/CleanupServer.java

```diff
@@ -95,6 +101,7 @@ public Boolean apply(String id) {
       }
 
       boolean serverDeleted = novaApi.getServerApi(regionAndId.getRegion()).delete(regionAndId.getId());
+      checkState(nodeTerminatedPredicate.apply(id), "server was not destroyed in the configured timeout");
```

Comment:
No real opinion on that to be honest!   I would say though, that if the `cleanupServer.apply()` is being invoked as per the link above (inside `cleanUpIncidentalResourcesOfDeadNodes`), then you probably want to avoid the exception here, as otherwise the first exception when iterating over `deadNodes` will prevent any remaining nodes even being attempted.